File: brain/db/supabase_client.py
"""
Supabase client connection.
"""
import os
from supabase import create_client, Client
from config import get_settings


def get_supabase_client() -> Client:
    """
    Create and return a Supabase client.
    Uses service role for backend operations (bypasses RLS).
    """
    settings = get_settings()
    return create_client(
        settings.supabase_url,
        settings.supabase_anon_key
    )


# Singleton instance
_supabase_client: Client | None = None


def get_supabase() -> Client:
    """Get or create Supabase client singleton."""
    global _supabase_client
    if _supabase_client is None:
        _supabase_client = get_supabase_client()
    return _supabase_client

File: frontend/src/supabaseClient.js
import { createClient } from '@supabase/supabase-js'

const supabaseUrl = 'https://nuuqkwdopdsgokiziyum.supabase.co'
const supabaseAnonKey = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Im51dXFrd2RvcGRzZ29raXppeXVtIiwicm9sZSI6ImFub24iLCJpYXQiOjE3ODQ5MDYwNTgsImV4cCI6MjEwMDQ4MjA1OH0.kuGd10dtw0mfAkB05LNV7ybkyZO5Q_q99wIkJBrn5Ic'

export const supabase = createClient(supabaseUrl, supabaseAnonKey)

